rust/azure_iot_operations_connector/src/base_connector/managed_azure_device_registry.rs

```diff
@@ -1572,6 +1494,37 @@ impl DatasetClient {
     pub fn device_status(&self) -> DeviceEndpointStatus {
         (*self.device_status.read().unwrap()).clone()
     }
+
+    /// Helper function to update the specific dataset status within the asset status
+    fn update_dataset_status(
+        asset_status_to_update: &mut adr_models::AssetStatus,
+        dataset_name: &str,
+        dataset_status: Result<(), AdrConfigError>,
+    ) {
+        if let Some(curr_dataset_status) =
+            asset_status_to_update
+                .datasets
+                .as_mut()
+                .and_then(|datasets| {
+                    datasets
+                        .iter_mut()
```

Comment:
Does this need to be mutable?

Reply:
yep - because we then set `curr_dataset_status` that we get from it's error